packages/hardhat-core/src/internal/core/providers/accounts.ts

```diff
@@ -249,54 +246,114 @@ export class LocalAccountsProvider extends ProviderWrapperWithChainId {
     chainId: number,
     privateKey: Buffer
   ): Promise<Uint8Array> {
-    const { AccessListEIP2930Transaction, LegacyTransaction } = await import(
-      "@nomicfoundation/ethereumjs-tx"
+    const { bytesToHex, bytesToInt, bytesToBigInt } = await import(
+      "@ethereumjs/util"
     );
-
-    const { Common } = await import("@nomicfoundation/ethereumjs-common");
+    const { addr, Transaction } = await import("micro-eth-signer");
 
     const txData = {
       ...transactionRequest,
       gasLimit: transactionRequest.gas,
     };
 
-    // We don't specify a hardfork here because the default hardfork should
-    // support all possible types of transactions.
-    // If the network doesn't support a given transaction type, then the
-    // transaction it will be rejected somewhere else.
-    const common = Common.custom({ chainId, networkId: chainId });
+    const accessList = txData.accessList?.map(({ address, storageKeys }) => {
+      return {
+        address: addr.addChecksum(bytesToHex(address)),
+        storageKeys:
+          storageKeys !== null ? storageKeys.map((k) => bytesToHex(k)) : [],
+      };
+    });
+
+    const authorizationList = txData.authorizationList?.map(
+      ({ chainId: authChainId, address, nonce, yParity, r, s }) => {
+        return {
+          chainId: authChainId,
+          address: addr.addChecksum(bytesToHex(address)),
+          nonce,
+          yParity: bytesToInt(yParity),
+          r: bytesToBigInt(r),
+          s: bytesToBigInt(s),
+        };
+      }
+    );
+
+    let checksummedAddress;
+    if (txData.to === undefined || txData.to === null) {
```

Comment:
Note that, as in v3, we should fail if there's no address and there's no data either, because that's not supported by `micro-eth-signer`.

In theory it is supported by ethereum, but it's almost certainly a user error.